app/routes/articles.js

```diff
@@ -197,14 +203,14 @@ router.put('/:id', checkArticleOwnership, (req, res) => {
         }
 
         req.flash('success', 'Successfully updated your article!')
-        res.redirect('/articles/' + req.params.id)
+        res.redirect('/articles/' + encodedTitle)
     })
 })
 
 //DELETE Article Route
-router.delete('/:id', checkArticleOwnership, (req, res) => {
-    if(!req?.params?.id) return res.render('error', {code: '404', msg: 'Invalid Article ID'})
-    Article.deleteOne({_id: req.params.id}, err => {
+router.delete('/:title', checkArticleOwnership, (req, res) => {
+    if(!req?.params?.title) return res.render('error', {code: '404', msg: 'Invalid Article Title'})
```

Comment:
` if(!req?.params?.title)`
is it possible for params to not exist on req, or that req is undefined? And if not then no null coalesing is needed here at all.